src/main/java/randoop/main/GenInputsAbstract.java

```diff
@@ -1148,6 +1170,117 @@ private static boolean shouldOmitClass(String classname) {
     }
   }
 
+  /**
+   * Returns classes in the given package. Does not include classes in sub-packages. Ignores
+   * non-accessible classes.
+   *
+   * @param packageName a package name
+   * @param accessibility the accessibility predicate
+   * @return classes in package {@code packageName}
+   */
+  private static Set<@ClassGetName String> getClassnamesFromPackage(
+      String packageName, AccessibilityPredicate accessibility) {
+    Set<@ClassGetName String> classnames = new HashSet<>();
+
+    for (String path : Globals.getClassPath().split(File.pathSeparator)) {
+      File location = new File(path);
+      if (location.isFile() && location.getName().endsWith(".jar")) {
+        classnames.addAll(getClassesWithPackageFromJar(location, packageName, accessibility));
+      } else if (location.isDirectory()) {
+        classnames.addAll(getClassesWithPackageFromDirectory(location, packageName, accessibility));
+      }
+    }
+
+    return classnames;
+  }
+
+  /**
+   * Searches in the given directory for classes with the given package. Directory must be root
+   * directory containing class files, e.g. {@code build/classes/} for gradle projects or {@code
+   * target/classes/} for maven projects.
+   *
+   * @param directory a directory
+   * @param packageName a package name
+   * @param accessibility the accessibility predicate
+   * @return classes with the given package that were found in the specified directory
+   */
+  private static Set<@ClassGetName String> getClassesWithPackageFromDirectory(
+      File directory, String packageName, AccessibilityPredicate accessibility) {
+    String packageNameAsFile = packageName.replace(".", File.separator);
+    File packageDirectory =
+        new File(directory.getPath().concat(File.separator).concat(packageNameAsFile));
+    if (packageDirectory.exists() && packageDirectory.isDirectory()) {
+      Set<@ClassGetName String> classnames = new HashSet<>();
+      for (File file :
+          packageDirectory.listFiles(f -> f.isFile() && f.getName().endsWith(".class"))) {
+
+        String relativePath =
+            file.getPath()
+                .substring(directory.getAbsolutePath().length() + File.separator.length());
+        String internalForm = relativePath.substring(0, relativePath.length() - ".class".length());
+        @SuppressWarnings("signature") // classname must have @ClassGetName annotation to be added
+        // to classnames set, but we assign string without any annotations
+        @ClassGetName String classname = internalForm.replace(File.separator, ".");
```

Comment:
Please use [`Signatures.internalFormToClassGetName`](http://plumelib.org/reflection-util/api/org/plumelib/reflection/Signatures.html#internalFormToClassGetName-java.lang.String-) rather than performing string manipulation.  Then you won't need to suppress any warnings, either.

Reply:
I tried, and when searching in jars it works just fine, but this method does not work well in directories. I am using Windows and have '\' separator instead of '/', so it simply returns the same string. Should I update `Signatures.internalFormToClassGetName` method to replace '\' too or should I ensure string passed to it has correct separator? Also, there are `Signatures.classfilenameToBinaryName` and `Signatures.binaryNameToClassGetName` methods, that might work. I am not sure what solution will be correct in this situation. Thank you for your help.

Reply:
Thank you for pointing that out.  The variable name `internalForm` is misleading -- on Unix, that variable holds a string in internal form, but on Windows it does not.  I suggest changing that variable name to avoid confusion.

Reply:
I decided to use `Signatures.classfilenameToBinaryName` and `Signatures.binaryNameToClassGetName` methods. They perform all string and type manipulations needed, so `internalForm` variable and `@SuprressWarnings` annotations can be deleted.